File: test_holy_fucking_god.js
// 测试脚本 - 使用 "holy fucking god" 作为测试文本
const axios = require('axios');

// 配置axios
axios.defaults.baseURL = 'http://localhost:8000';
axios.defaults.timeout = 600000;

// 测试数据
const testText = 'holy fucking god';
const sourceLang = 'en';
const targetLang = 'zh';

async function runTests() {
  console.log('开始测试...');
  
  try {
    // 1. 测试处理文本
    console.log('1. 测试处理文本...');
    const processResponse = await axios.post('/api/process-text', {
      text: testText,
      source_language: sourceLang,
      target_language: targetLang
    });
    
    const fileId = processResponse.data.file_id;
    console.log(`获取到文件ID: ${fileId}`);
    
    // 2. 等待处理完成
    console.log('2. 等待处理完成...');
    let status = { status: 'processing' };
    while (status.status === 'processing') {
      await new Promise(resolve => setTimeout(resolve, 2000));
      status = await axios.get(`/api/status/${fileId}`);
      status = status.data;
      console.log(`处理状态: ${status.status}, 进度: ${status.progress}%`);
    }
    
    if (status.status === 'error') {
      throw new Error(`处理失败: ${status.error}`);
    }
    
    console.log('处理完成！');
    
    // 3. 测试获取词汇表
    console.log('3. 测试获取词汇表...');
    const vocabResponse = await axios.get(`/api/vocab/${fileId}`);
    const vocab = vocabResponse.data.vocab;
    console.log(`词汇表长度: ${vocab.length}`);
    vocab.forEach((word, index) => {
      console.log(`${index + 1}. ${word.word} - ${word.context_meaning}`);
    });
    
    // 4. 测试获取第一阶段学习进度
    console.log('4. 测试获取第一阶段学习进度...');
    const learningProgressResponse = await axios.get(`/api/learn/${fileId}/progress`);
    const learningProgress = learningProgressResponse.data;
    console.log(`第一阶段学习进度: 当前单元 ${learningProgress.current_unit + 1}/${learningProgress.total_units}`);
    learningProgress.units.forEach((unit, index) => {
      console.log(`单元 ${index + 1}: ${unit.word_count} 个单词, 完成: ${unit.completed}`);
    });
    
    // 5. 测试单词学习（模拟学完所有单词）
    console.log('5. 测试单词学习...');
    let currentIndex = 0;
    while (currentIndex < vocab.length) {
      // 模拟学习单词
      const nextWordResponse = await axios.post(`/api/learn/${fileId}/next-word`);
      currentIndex = nextWordResponse.data.new_index;
      console.log(`学习进度: ${currentIndex}/${vocab.length}`);
      
      // 检查学习进度
      const progressResponse = await axios.get(`/api/learn/${fileId}/progress`);
      const progress = progressResponse.data;
      console.log(`当前单元: ${progress.current_unit + 1}, 完成状态: ${progress.units.map(u => u.completed ? '✓' : '✗').join(', ')}`);
    }
    
    // 6. 测试检查覆盖度
    console.log('6. 测试检查覆盖度...');
    const coverageResponse = await axios.get(`/api/learn/${fileId}/check-coverage`);
    const coverage = coverageResponse.data;
    console.log(`覆盖度检查结果: 可以组成句子? ${coverage.can_form_sentences}, 单元完成? ${coverage.unit_completed}`);
    
    // 7. 测试生成句子翻译题
    console.log('7. 测试生成句子翻译题...');
    try {
      const quizResponse = await axios.get(`/api/learn/${fileId}/sentence-quiz`);
      console.log(`翻译题: ${quizResponse.data.original_sentence}`);
      console.log(`正确翻译: ${quizResponse.data.correct_translation}`);
      console.log(`正确tokens: ${quizResponse.data.correct_tokens}`);
    } catch (quizError) {
      if (quizError.response && quizError.response.status === 404 && quizError.response.data.detail === 'No more eligible sentences') {
        console.log('所有句子都已使用，单元已完成');
      } else {
        throw quizError;
      }
    }
    
    // 8. 测试获取阶段一和阶段二的单元列表
    console.log('8. 测试获取阶段一和阶段二的单元列表...');
    const phase1UnitsResponse = await axios.get(`/api/${fileId}/phase/1/units`);
    const phase1Units = phase1UnitsResponse.data.units;
    console.log('阶段一单元:');
    phase1Units.forEach((unit, index) => {
      console.log(`单元 ${index + 1}: ${unit.word_count} 个单词, 完成: ${unit.completed}`);
    });
    
    const phase2UnitsResponse = await axios.get(`/api/${fileId}/phase/2/units`);
    const phase2Units = phase2UnitsResponse.data.units;
    console.log('阶段二单元:');
    phase2Units.forEach((unit, index) => {
      console.log(`单元 ${index + 1}: ${unit.sentences_count} 个句子, 完成: ${unit.completed}`);
    });
    
    // 9. 测试阶段二练习
    console.log('9. 测试阶段二练习...');
    if (phase2Units.length > 0) {
      const unitId = 0;
      try {
        const exerciseResponse = await axios.get(`/api/${fileId}/phase/2/unit/${unitId}`);
        console.log(`练习类型: ${exerciseResponse.data.exercise_type}`);
        console.log(`练习数据:`, exerciseResponse.data.data);
        
        // 测试进入下一个练习
        const nextExerciseResponse = await axios.post(`/api/${fileId}/phase/2/unit/${unitId}/next`);
        console.log(`下一个练习:`, nextExerciseResponse.data);
      } catch (exerciseError) {
        if (exerciseError.response && exerciseError.response.data.unit_complete) {
          console.log('单元已完成');
        } else {
          throw exerciseError;
        }
      }
    }
    
    console.log('测试完成！');
  } catch (error) {
    console.error('测试失败:', error);
  }
}

runTests();

File: test_holy_fucking_god.py
#!/usr/bin/env python3
"""Test the complete flow for 'holy fucking god'"""
import requests
import time
import json

BASE_URL = "http://localhost:8000"
TEST_TEXT = "holy fucking god"

def test_complete_flow():
    """Test the complete flow from text processing to phase 2 exercises"""
    print("Testing complete flow for:", TEST_TEXT)
    print("=" * 60)
    
    # Step 1: Process text
    print("Step 1: Processing text...")
    response = requests.post(
        f"{BASE_URL}/api/process-text",
        json={"text": TEST_TEXT}
    )
    assert response.status_code == 200
    file_id = response.json().get("file_id")
    print(f"✓ Text processed successfully, file_id: {file_id}")
    
    # Wait for processing to complete
    print("Waiting for processing to complete...")
    for i in range(60):
        time.sleep(1)
        status = requests.get(f"{BASE_URL}/api/status/{file_id}").json()
        if status.get("status") == "completed":
            print("✓ Processing completed successfully")
            break
    
    # Step 2: Check LLM output format
    print("Step 2: Checking LLM output format...")
    sentences = requests.get(f"{BASE_URL}/api/sentences/{file_id}").json().get("sentences", [])
    assert len(sentences) > 0
    
    sentence_data = sentences[0]
    translation_result = sentence_data.get("translation_result", {})
    
    # Check tokenized_translation format
    tokenized_translation = translation_result.get("tokenized_translation", "")
    assert "/" not in tokenized_translation, f"tokenized_translation should not contain /: {tokenized_translation}"
    print(f"✓ tokenized_translation format correct: {tokenized_translation}")
    
    # Check translation array
    translation = translation_result.get("translation", [])
    assert isinstance(translation, list)
    assert len(translation) > 0
    for token in translation:
        assert "translation" in token
        assert "/" not in token["translation"], f"translation should not contain /: {token['translation']}"
    print("✓ translation array format correct")
    
    # Step 3: Check vocabulary format
    print("Step 3: Checking vocabulary format...")
    vocab = requests.get(f"{BASE_URL}/api/vocab/{file_id}").json().get("vocab", [])
    assert len(vocab) > 0
    
    for entry in vocab:
        # Check required fields are present
        assert "word" in entry
        assert "tokens" in entry
        assert "variants" in entry
        assert "ipa" in entry
        assert "context_meaning" in entry
        assert "examples" in entry
        assert "options" in entry
        assert "grammar" in entry
        assert "morphology" in entry
        
        # Check no redundant translation field
        assert "translation" not in entry, "dictionary entries should not have translation field"
    print("✓ Vocabulary format correct")
    
    # Step 4: Test phase 2 exercises
    print("Step 4: Testing phase 2 exercises...")
    
    # Reset progress
    reset_response = requests.post(
        f"{BASE_URL}/api/{file_id}/phase/2/set-progress",
        json={"unit_id": 0, "exercise_index": 0, "exercise_type_index": 0}
    )
    assert reset_response.status_code == 200
    
    # Get exercises
    exercise_types = []
    for i in range(4):
        exercise_response = requests.get(f"{BASE_URL}/api/{file_id}/phase/2/unit/0")
        assert exercise_response.status_code == 200
        
        exercise_data = exercise_response.json()
        if exercise_data.get("unit_complete"):
            break
        
        exercise_type = exercise_data.get("exercise_type")
        exercise_types.append(exercise_type)
        print(f"  Exercise {i+1}: {exercise_type}")
        
        # Move to next
        next_response = requests.post(f"{BASE_URL}/api/{file_id}/phase/2/unit/0/next")
        assert next_response.status_code == 200
    
    # Verify both exercise types
    assert "masked_sentence" in exercise_types
    assert "translation_reconstruction" in exercise_types
    print("✓ Both exercise types present")
    
    # Verify no repetition
    for i in range(1, len(exercise_types)):
        assert exercise_types[i] != exercise_types[i-1], f"Exercise type repeated: {exercise_types[i]}"
    print("✓ No exercise repetition")
    
    print("=" * 60)
    print("✅ All tests passed! The complete flow is working correctly.")

if __name__ == "__main__":
    test_complete_flow()
